File: src/main/java/be/superjoran/mint/services/CategoryServiceImpl.java
package be.superjoran.mint.services;

import be.superjoran.common.model.DomainObjectCrudServiceSupport;
import be.superjoran.mint.dao.CategoryDao;
import be.superjoran.mint.domain.Category;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Service;

@Service
public class CategoryServiceImpl extends DomainObjectCrudServiceSupport<Category> implements CategoryService {
    private final CategoryDao categoryDao;

    public CategoryServiceImpl(CategoryDao categoryDao) {
        this.categoryDao = categoryDao;
    }

    @Override
    protected CrudRepository<Category, String> getDao() {
        return this.categoryDao;
    }
}

File: src/main/java/be/superjoran/mint/services/CategoryServiceImpl.kt
package be.superjoran.mint.services

import be.superjoran.common.model.DomainObjectCrudServiceSupport
import be.superjoran.mint.dao.CategoryDao
import be.superjoran.mint.domain.Category
import org.springframework.stereotype.Service
import javax.validation.Validator

@Service
class CategoryServiceImpl(
        override val dao: CategoryDao,
        override val validator: Validator
) : DomainObjectCrudServiceSupport<Category>(), CategoryService
